projects/step-frontend/src/lib/angularjs/js/app.js

```diff
@@ -959,27 +959,9 @@ angular
             return $scope.tableHandle.getSelectedIds();
           };
 
-          $scope.handle.getFilter = function () {
-            const searchQuery = {};
-            for (const i in $scope.tableHandle.columns()[0]) {
-              let oAjaxData = $scope.tableHandle.columns(i).context[0].oAjaxData;
-
-              let column = oAjaxData.columns[i];
-              let order = oAjaxData.order[0];
-
-              searchQuery['columns[' + i + '][data]'] = column.data;
-              searchQuery['columns[' + i + '][name]'] = column.name;
-              searchQuery['columns[' + i + '][searchable]'] = column.searchable || true;
-              searchQuery['columns[' + i + '][orderable]'] = column.orderable || true;
-              searchQuery['columns[' + i + '][search][value]'] = column.search.value;
-              searchQuery['columns[' + i + '][search][regex]'] = column.search.regex;
-              searchQuery['order[0][column]'] = (order && order.column) || 0;
-              searchQuery['order[0][dir]'] = (order && order.dir) || 'asc';
-              searchQuery['draw'] = (oAjaxData && oAjaxData.draw) || 1;
-              searchQuery['start'] = (oAjaxData && oAjaxData.start) || 0;
-              searchQuery['length'] = (oAjaxData && oAjaxData.length) || 10;
-            }
-            return searchQuery;
+          $scope.handle.getFilters = function () {
+            const lastRequest = $scope.tableHandle.getLastServerSideRequest();
```

Comment:
@dvladir I assume this works differently with the new tables i.e. that the filters are rebuilt each time

Reply:
@jeromecomte 
As assign entities is based on legacy tables, I have to modify the table's logic to store the last request made by table and then extract filters from it, when it is required.

In case of new tables, we already have the similar functionality that allows to retrieve the request object. But you're right it works a bit differently - table stores filters and search values for different columns. By invoking table's method `getTableFilterRequest` - it builds the request object, based on stored values.